design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+   "admin-username": "zappa"
+}
+```
+
+##### Example `bash` Secrets File
+
+```
+   export URL="https://database.example.com"
+   export ADMIN_PASSWORD="p@$$w0rd"
+   export ADMIN_USERNAME="zappa"
+```
+
+##### Example `dotenv` (.env) Secrets File
+
+```
+   URL="https://database.example.com"
+   ADMIN_PASSWORD="p@$$w0rd"
+   ADMIN_USERNAME="zappa"
+```
+
+### Upgrading Existing Secrets Provider Deployments
+
+At a high level, converting an existing Secrets Provider deployment to use
+annotation-based configuration and/or push-to-file mode:
+
+- Inspect the existing application Deployment manifest (if available) or
+  use `kubectl edit` to inspect the application Deployment.
+- Convert the Service Provider container/Conjur environment variable settings
+  to the equivalent annotation-based setting. Retain the `K8S_SECRETS`
+  setting for now.
+- If you are using the Secrets Provider as an init container, and you would
+  like to convert from K8s Secrets mode to push-to-file mode:
+  - Add push-to-file annotations:
+    - For each existing Kubernetes Secret, you may wish to create a separate
+      secrets group for push-to-file.
+    - `conjur.org/conjur-secrets.{group}`: Inspect the manifests for the
+      existing Kubernetes Secret(s). The manifests should contain a
+      `stringData` section that contains secrets key/value pairs.
+      Map the `stringData` entries to a YAML list value for conjur-secrets,
+      using the secret names as aliases.
+    - `conjur.org/secret-file-path.{group}`: Configure a target location
+    - `conjur.org/secret-file-type.{group}`: Configure a desired type,
+      depending on how the application will consume the secrets file.
+    - `conjur.org/secret-file-perms.{group}`: Determine desired file
+      permissions (see [Security](#security) section).
+  - Add Pod `securityContext` (see [Security](#security) section).
+  - Delete existing Kubernetes Secrets or their manifests:
+    - If using Helm, delete Kubernetes Secrets manifests and do a
+      `helm upgrade ...`
+    - Otherwise, `kubectl delete ...` the existing Kubernetes Secrets
+  - Delete the `K8S_SECRETS` environment variable setting from the application
+    Deployment (or its manifest).
+  - Modify application to consume secrets as files:
+    - Modify application to consume secrets files directly, or...
+    - Modify the Deployment's spec for the app container so that the
+      `command` entrypoint includes sourcing of a bash-formatted secrets file.
+
+### Project Scope and Limitations
+
+The initial implementation and testing will be limited to:
+
+-   Authentication containers to be tested:
+
+    -   Secrets Provider init container
+    -   Secrets Provider standalone Pod
+
+-   Platforms:
+
+    -   Kubernetes (this will be either Kubernetes-in-Docker, or GKE).
+    -   OpenShift 4.6
+
+#### Out of Scope
+
+- For this release, configuration will only be read upon startup.
+- For this release, annotations are required to have a group.
```

Comment:
Maybe be clear that you are only talking about the "push to file" annotations, and even link to the "Annotations for Push-to-File Secrets Mappings" section?